api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -36,7 +37,7 @@
  * href=https://github.com/w3c/distributed-tracing>w3c/distributed-tracing</a>.
  */
 @Immutable
-public class HttpTraceContext implements HttpTextFormat<SpanContext> {
+public class HttpTraceContext implements HttpTextFormat {
```

Comment:
The name `HttpTraceContext` had already been confusing before this PR but now is even more so. I suggest renaming to `W3cTraceContextFormat` or something similar. But since this PR is already huge, maybe do it in a separate PR (that *could* be merged before this one to make it less confusing 😃)

```suggestion
public class W3cTraceContextFormat implements HttpTextFormat {
```

Reply:
Let's definitely create an issue to consider our options here.

Reply:
Hey @Oberon00 please open a new issue about this on the Specification or in this repo to follow up.